csharp/ql/test/library-tests/csharp9/record.expected

```diff
@@ -0,0 +1,222 @@
+types
+| Record.cs:9:1:15:1 | Person | IEquatable<Person> |
+| Record.cs:17:1:23:1 | Teacher | IEquatable<Teacher> |
+| Record.cs:25:1:30:1 | Student | IEquatable<Student> |
+| Record.cs:32:1:32:57 | Person1 | IEquatable<Person1> |
+| Record.cs:34:1:35:35 | Teacher1 | IEquatable<Teacher1> |
+| Record.cs:37:1:38:35 | Student1 | IEquatable<Student1> |
+| Record.cs:40:1:44:1 | Pet | IEquatable<Pet> |
+| Record.cs:46:1:57:1 | Dog | IEquatable<Dog> |
+members
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.Equals(object) | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.Equals(object, object) | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.GetHashCode() | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.GetType() | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.MemberwiseClone() | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.Object() | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.ReferenceEquals(object, object) | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.ToString() | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Object.~Object() | no location |
+| Record.cs:6:18:6:31 | IsExternalInit | System.Runtime.CompilerServices.IsExternalInit.IsExternalInit() | no location |
+| Record.cs:9:1:15:1 | Person | Person.!=(Person, Person) | Record.cs:9:15:9:20 |
+| Record.cs:9:1:15:1 | Person | Person.<Clone>$() | no location |
+| Record.cs:9:1:15:1 | Person | Person.==(Person, Person) | Record.cs:9:15:9:20 |
+| Record.cs:9:1:15:1 | Person | Person.EqualityContract | Record.cs:9:15:9:20 |
+| Record.cs:9:1:15:1 | Person | Person.Equals(Person) | no location |
+| Record.cs:9:1:15:1 | Person | Person.Equals(object) | no location |
+| Record.cs:9:1:15:1 | Person | Person.FirstName | Record.cs:12:19:12:27 |
+| Record.cs:9:1:15:1 | Person | Person.GetHashCode() | no location |
+| Record.cs:9:1:15:1 | Person | Person.LastName | Record.cs:11:19:11:26 |
+| Record.cs:9:1:15:1 | Person | Person.Person(Person) | no location |
```

Comment:
When a copy ctor has no location, it's a compiler generated member. We have no body for these, but we know the exact behavior of these, we could add flow summaries for these ctors.

The same is true for generated copy constructors that are extracted from assemblies, but I'm unsure if we can figure out from the signature and metadata if a ctor is generated or not.